components/eam/src/chemistry/modal_aero/modal_aero_coag.F90

```diff
@@ -1,2337 +1,1363 @@
-!               qaer_tmpb(iaer,nacc) = qaer_tmpb(iaer,nacc) + tmp_dq
-!            end do
-!         end if
-!      end if
-
-! mass transfer out of marine-organics accum mode
-!    uses average number conc. for accum and pcarbon modes
-      if (nmacc > 0) then
-         tmp1 = max( 0.0_r8, ybetaij3( 9)*qnum_tmpc(nacc) )
-         tmp2 = max( 0.0_r8, ybetaij3(10)*qnum_tmpc(npca) )
-         tmpa = tmp1 + tmp2
-         tmpc = deltat*tmpa
-         if (tmpc > epsilonx2) then
-            ! calc coag change only when it is not ~= zero
-            tmp_xf = 1.0_r8 - exp(-tmpc)
-            tmp2 = tmp2/tmpa
-            tmp1 = 1.0_r8 - tmp2
-            do iaer = 1, naer
-               tmp_dq = tmp_xf*qaer_tmpa(iaer,nmacc)
-               qaer_tmpb(iaer,nmacc) = qaer_tmpb(iaer,nmacc) - tmp_dq
-               qaer_tmpb(iaer,nacc ) = qaer_tmpb(iaer,nacc ) + tmp_dq*tmp1
-               qaer_tmpb(iaer,npca ) = qaer_tmpb(iaer,npca ) + tmp_dq*tmp2
-               qaer_del_coag_in(iaer,i_agepair_pca ) &
-                                    = qaer_del_coag_in(iaer,i_agepair_pca ) + tmp_dq*tmp2
-            end do
-         end if
-      end if
+    ! Free-molecular form: equation h.7b of whitby et al. (1991)
 
-! mass transfer out of pcarbon mode 
-!    uses average number conc. for accum mode
-      if (npca > 0) then
-         tmpc = max( 0.0_r8, ybetaij3(2)*qnum_tmpc(nacc) )
-         tmpc = deltat*tmpc
-         if (tmpc > epsilonx2) then
-            tmp_xf = 1.0_r8 - exp(-tmpc)
-            do iaer = 1, naer
-               tmp_dq = tmp_xf*qaer_tmpa(iaer,npca)
-               qaer_tmpb(iaer,npca) = qaer_tmpb(iaer,npca) - tmp_dq
-               qaer_tmpb(iaer,nacc) = qaer_tmpb(iaer,nacc) + tmp_dq
-            end do
-         end if
-      end if
+    coagfm3 = kfmatac * sqdgat7 * bm3i( n1, n2n, n2a ) * (   &
+      esat49   &
+      +  r * esat36  * esac01   &
+      + two * r2 * esat25  * esac04   &
+      + rx4 * esat09  * esac16   &
+      + ri3 * esat100 * esac09   &
+      + two * ri1 * esat64  * esac01 )
 
-! mass transfer out of accum mode - there is no transfer out of this mode
+    ! Gain by accumulation mode = loss from aitken mode. Harmonic mean
 
-#if ( defined( CAMBOX_ACTIVATE_THIS ) )
-      if ( ldiag15n ) then
-! diagnostics
-      do n = 1, 3
-         write(lun15n,'(a,i5,1p,10e11.3)') 'n, qnum_tmpa/b/c', n, &
-         qnum_tmpa(n), qnum_tmpc(n), qnum_tmpb(n), &
-         qnum_tmpb(n)-qnum_tmpa(n), &
-         1.0_r8-qnum_tmpb(n)/max(1.0e-5_r8,qnum_tmpa(n))
-      end do
-      do n = 1, 3
-         write(lun15n,'(a,i5,1p,10e11.3)') 'n, dgnd/w, densw', n, &
-         dgn_a(n), dgn_awet(n), wetdens(n)
-      end do
-      end if
-#endif
+    qv12 = coagnc3 * coagfm3 / ( coagnc3 + coagfm3 )
 
-      qnum_cur = qnum_tmpb
-      qaer_cur = qaer_tmpb
+    !-----------------------------------------------------
+    ! Intramodal coagulation
+    !-----------------------------------------------------
```

Comment:
The following calculations for aitken and acumulation modes are very similar. Can we create a subroutine and call it twice (for aitken and accumulation)with different inputs ?

Reply:
Good point - and done :-)